modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -145,7 +145,21 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint>> extends Psbt {
 
     updates.forEach((update, index) => {
       psbt.updateInput(index, update);
-      psbt.updateInput(index, { witnessUtxo: prevOutputs[index] });
+      psbt.updateInput(index, { witnessUtxo: { script: prevOutputs[index].script, value: prevOutputs[index].value } });
+    });
+
+    // Remove unknownKeyVals from ins and outs if they are empty.
+    // We do this because we want to have the psbt be changeless from the transaction it is wrapping.
```

Comment:
Not sure I understand this reasoning. An empty list-map is equivalent to it not existing in the PSBT serialization format.

Reply:
It's true - but when you deserialize a psbt hex, these empty arrays do not show up in the `unspents` and `outputs`. If you were to do a comparison of 

```typescript
...
const psbtHex = originalPsbt.toHex();
const deserializedPsbt = UtxoPsbt.fromHex(psbtHex, { network });
assert.deepStrictEqual(deserializedPsbt, originalPsbt)
```
This would fail because of these empty `unknownKeyVals`

Reply:
Hmm, that seems like something to fix in test code rather than here in production code then.

Reply:
Or something to fix in a better written comparison (for example, one way to compare PSBTs is to serialize them to a buffer and compare the buffers, since _that_ is a canonical representation).